apis/vshn/v1/dbaas_vshn_keycloak.go

```diff
@@ -120,6 +120,17 @@ type VSHNKeycloakServiceSpec struct {
 	// The themes need to be be placed in the `/themes` directory of the custom image.
 	// the providers need to be placed in the `/providers` directory of the custom image.
 	CustomizationImage VSHNKeycloakCustomizationImage `json:"customizationImage,omitempty"`
+
+	// Configuration can be used to provide a configmap containing configurations for the
+	// keycloak instance. The config is a JSON file based on the keycloak export files.
+	// The referenced configmap, must have the configuration in a field called `keycloak-config.json`
+	Configuration *string `json:"configuration,omitempty"`
+
+	// CustomEnvVariables can be used to provide custom environment variables from a
+	// provided secret for the keycloak instance. The environment variables provided
+	// can for example be used in the custom JSON configuration provided in the `Configuration`
+	// field with `$(env:<ENV_VAR_NAME>:-<some_default_value>)`
+	CustomEnvVariables *string `json:"customEnvVariables,omitempty"`
```

Comment:
I would add `SecretRef` to this name.

Reply:
I thought about this as well, but then the behaviour would be inconsistent. I would either leave it as is, or change both to refs (SecretReference and LocalObjectReference respectively)

Reply:
I would actually change the naming of the variables, so it's consistent with what we already have for the alertmanager configs: https://github.com/vshn/appcat/blob/master/apis/vshn/v1/common_types.go#L159

So everything that's a reference should contain `ref`. I would even go so far as to call the other field `CustomConfigurationTemplateRef` because it's a template.

Reply:
I've renamed the fields to include the `ref` part.
However, I disagree with the `CustomConfigurationTemplateRef`. This is not really a template, but a configuration itself.

Reply:
I mean it kinda is, you can add variables that get replaced by actual values. Sounds very templaty to me :)

But I see that it might get too long then. So it's fine for me.